src/affinity.cc

```diff
@@ -0,0 +1,83 @@
+// non-platform-specific code for thread pinning
+
+#include "affinity.h"
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <libenvpp/env.hpp>
+
+namespace celerity::detail::thread_pinning {
+
+std::string thread_type_to_string(const thread_type t_type) {
+	switch(t_type) {
+	case thread_type::application: return "application";
+	case thread_type::scheduler: return "scheduler";
+	case thread_type::executor: return "executor";
+	default: break;
+	}
+	if(t_type >= thread_type::first_backend_worker && t_type < thread_type::first_host_queue) {
+		return fmt::format("backend_worker_{}", t_type - thread_type::first_backend_worker);
+	}
+	if(t_type >= thread_type::first_host_queue && t_type < thread_type::max) { return fmt::format("host_queue_{}", t_type - thread_type::first_host_queue); }
+	return fmt::format("unknown({})", static_cast<uint32_t>(t_type));
+}
+
+namespace {
+	// When we no longer need to support compilers without a working std::views::split, get rid of this function
+	std::vector<std::string> split(const std::string_view str, const char delim) {
+		std::vector<std::string> result;
+		size_t start = 0;
+		size_t end = str.find(delim);
+		while(end != std::string::npos) {
+			result.push_back(std::string(str.substr(start, end - start)));
+			start = end + 1;
+			end = str.find(delim, start);
+		}
+		if(start < str.size()) result.push_back(std::string(str.substr(start)));
+		return result;
+	}
+} // namespace
+
+environment_configuration parse_validate_env(const std::string_view str) {
```

Comment:
It was when I started implementing it, but I purposefully moved it since I think it's better for this level of parsing to be in the module it interacts with rather than the generic config file (where it detracts from that code's main purpose).

I don't think including libenvpp in two TUs is a problem, we're not talking about a header.